content/en/docs/setup/production-environment/windows/user-guide-windows-containers.md

```diff
@@ -218,7 +218,10 @@ Here are values used today for each Windows Server version.
 ### Simplifying with RuntimeClass
 
 [RuntimeClass] can be used to simplify the process of using taints and tolerations. 
-A cluster administrator can create a `RuntimeClass` object which is used to encapsulate these taints and tolerations.
+A cluster administrator can create a `RuntimeClass` object which is used to encapsulate these taints and tolerations. 
+Please note that this is the recommended way to schedule pods onto Windows nodes instead of using nodeSelector and
+tolerations as they can be added to pod spec by any user. In future, we will enforce runtimeclasses to be used. Pods
+having just nodeSelector and tolerations are going to be rejected during apiserver admission.
```

Comment:
I suggest calling this out earlier under `Taints and Tolerations`. Otherwise the reader is going think this might come in only if they are using multiple Windows versions in the same cluster. IOW, move `RuntimeClasses` to the same level as `Taints and Tolerations` and reword appropriately.

Reply:
Agree to move `RuntimeClasses` to be a peer of `Taints and Tolerations`

Reply:
Please remove the "In future, ..." statement. We don't predict/document future.

Reply:
Here's the [style guide](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future) on future statements.

Maybe the heading should change? RuntimeClass stops being a simplification and is instead the primary, recommended way to define the OS-level scheduling constraints for Pods.

Reply:
Reorganized the file to bring in some more clarity. PTAL

@tengqm  @sftim - This is sort of a deprecation notice. So, it should be good as per the link you mentioned above?

Reply:
We shouldn't put deprecation notices only in the “Getting started” section - which is where this page currently lives. If (and only if) the deprecation is a firm commitment, I recommend documenting that deprecation clearly _and_ mentioning it in the v1.22 release notes - perhaps open one or more issues about that and open a separate docs PR as well.

(I'd also want to check with SIG Architecture about this deprecation; people currently use `nodeSelector` quite a bit).

Reply:
> I'd also want to check with SIG Architecture about this deprecation; people currently use nodeSelector quite a bit).

This is not for nodeSelector field in general but for Windows pods specifically. 

> perhaps open one or more issues about that and open a separate docs PR as well.

You mean opening release notes PR and a github issue for it?

Reply:
I don't think should say anything about `In future, we will enforce runtimeclasses to be used. Pods
having just nodeSelector and tolerations are going to be rejected during apiserver admission.` In this PR.

We should get approval of https://github.com/kubernetes/enhancements/pull/2803 from all participating SIGs before saying anything definitive.

Also - because of this maybe we shouldn't say using runtime classes are recommended over nodeSelector generally.
If we can provide specific reasons why it is better (that we can commit to) great. 
If not should we just document it as an alternative right now?

Reply:
My take:
RuntimeClass is the better mechanism, whatever OS your container image expects. It's fine to document that it's the approach we recommend.

We shouldn't document the deprecation from enhancement 2803 until it has been accepted and progressed appropriately. We shouldn't say anything about the other mechanism being unacceptable or deprecated.

Reply:
> I don't think should say anything about In future, we will enforce runtimeclasses to be used. Pods having just nodeSelector and tolerations are going to be rejected during apiserver admission. In this PR.

> We should get approval of kubernetes/enhancements#2803 from all participating SIGs before saying anything definitive.

That's a fair argument. I'll remove that statement. 


> Also - because of this maybe we shouldn't say using runtime classes are recommended over nodeSelector generally.
If we can provide specific reasons why it is better (that we can commit to) great.
If not should we just document it as an alternative right now?


I think the argument is other way around, we know that the runtimeclasses are better, so we are proposing to make them authoritative in the linked enhancement. 

I'll go with @sftim's suggestion that using runtimeclasses is recommended and better.

Reply:
Updated. PTAL

Reply:
I more meant that we shouldn't just say they are better but we should list some reasons why they are better.
Right now nodeSelectors are ubiquitous for identifying Windows pods I can see a lot of people asking why they should switch. 
Since the KEP isn't approved and we cannot say nodeSelectors won't work in the future can we give some examples of why people should use them today?

Reply:
I tried doing that [here](https://github.com/kubernetes/website/pull/29062/files#diff-631d5a315dd095196e6bd6429528876846b7b9fb918d87caae10c23427ac5ad7R137). I also mentioned that runtimeclasses are created by cluster admin. That is the reason for the switch.  Do you want me to add more? Something from historical context that runtimeclasses didn't exist back then...

Reply:
As you also need to be a cluster admin to register a Windows node / configure automation to allow admission, _and_ this is the production Windows setup guide, I think it's not only OK to recommend using a RuntimeClass for scheduling, I strongly think we should.